src/framework/theme/components/checkbox/_checkbox.component.theme.scss

```diff
@@ -6,71 +6,102 @@
 
 @mixin nb-checkbox-theme() {
   @mixin hover-checked-border($color) {
-    .custom-control-input:hover ~ .custom-control-indicator,
-    .custom-control-input:checked ~ .custom-control-indicator,
-    .custom-control-input:hover:checked ~ .custom-control-indicator {
-      border-color: $color;
+    .customised-control-input:hover ~ .customised-control-indicator,
+    .customised-control-input:checked ~ .customisedised-control-indicator,
+    .customised-control-input:hover:checked ~ .customised-control-indicator {
+      //border-color: $color;
+      // TODO implement hover variables
     }
   }
 
   nb-checkbox {
 
-    .custom-control {
+    .customised-control {
+      position: relative;
+      display: inline-flex;
       margin: 0;
       min-height: inherit;
       padding: 0.375rem 1.5rem;
+      line-height: 0;
     }
 
-    .custom-control-indicator {
-      position: absolute;
+    .customised-control-indicator {
       background-color: nb-theme(checkbox-bg);
       width: nb-theme(checkbox-size);
       height: nb-theme(checkbox-size);
       border: nb-theme(checkbox-border-size) solid nb-theme(checkbox-border-color);
-      top: 50%;
+
       transform: translateY(-50%);
+      border-radius: 0.25rem;
+      position: relative;
+      top: 50%;
+      display: flex;
+      justify-content: center;
+
+      &::before {
```

Comment:
let's move before implementation into scss mixin so that we don't have so many repeats